.github/workflows/e2e-arm64-template.yaml

```diff
@@ -12,7 +12,8 @@ jobs:
     # this is to prevent the job to run at forked projects
     if: github.repository == 'etcd-io/etcd'
     runs-on: [self-hosted, Linux, ARM64]
-    container: golang:1.20-bullseye
+    container:
+      image: golang:bullseye@sha256:02f350d8452d3f9693a450586659ecdc6e40e9be8f8dfc6d402300d87223fdfa
```

Comment:
```suggestion
      image: golang:1.20.7-bullseye
```

Reply:
also suggest to 
- get it defined in a centralised file, but can be in a separate PR.
- Keep consistent with the go version defined in `.go-version`.

Reply:
?? Do you understand that `golang:1.20.7-bullseye` is not a static image? It's a channel of latest debian bullseye release with golang 1.20.7 installed on top of it. We don't use Go version from it, only the tools that are taken from bullseye debian. Those tools can get patches so not specifying sha is not doing anything for us.

Reply:
For the image name, the name can be `repo:tag@digest`. When digest is in the name, the tag will be ignore during pulling. But the tag is alias for digest and it's easy to read. For the pin image, I think digest is good choice.

Reply:
> From our point of view (just using image tools) golang:bullseye and golang:1.20.7-bullseye are the same.

Except for the point I made above about debugging locally where it is convenient for go versions to align to the version the project uses.

I tend to agree with @ahrtr that we should aim to be consistent with our `.go-version`.  We can possibly even re-use that variable for this image definition.

Reply:
This discussion is derailing, as mentioned above, the only reason we are using `golang` image is fact that it has tools like `git` pre installed. I strongly oppose a dependency on `.go-version`.

Reply:
Not intending to derail.  Just trying to explain that there there are situations I was in when completing our `arm64` work where having an image that also contained a golang install (and the same version etcd used) was invaluable. 

When CI was failing on `arm64` I could start a container manually on the `arm64` runner and step through commands CI was running to do additional debug on what was going wrong, without having to manually install a the required go version in the container first.

If we merge this, we can still do that debugging when needed we just need to know to `docker run ...` the right version of the image, instead of the image that will now be present in the github actions workflows.

As mentioned above, I have approved the changes and this isn't a blocker from my perspective. However if you are asking for my opinion, I would prefer to have consistent versions across our codebase and it would be one less thing to think about when doing debugging on `arm64` runners if we keep the versions aligned.